src/components/APIDoc/ApiDoc.tsx

```diff
@@ -43,23 +43,34 @@ export const ApiDoc: React.FunctionComponent<ApiDocProps> = props => {
         { openapi.servers && (
             <StackItem>
                 <ServerList servers={openapi.servers}/>
+            </StackItem>
+
+        )}
+        { openapi.components?.securitySchemes && (
+            <StackItem className="pf-u-pb-lg">
                 <Divider
-                    className="apid-c-divider pf-u-pt-xl"
+                    className="apid-c-divider pf-u-pb-md"
                     inset={{default: 'insetNone',}}
                 />
+                <SecuritySchemeList schemes={Object.values(openapi.components.securitySchemes).map(s => deRef(s, openapi))} />
             </StackItem>
-
         )}
-        { openapi.components?.securitySchemes && (
+        
+        { paths.loading ? <Bullseye><Spinner /></Bullseye> : 
             <StackItem>
-                <SecuritySchemeList schemes={Object.values(openapi.components.securitySchemes).map(s => deRef(s, openapi))} />
                 <Divider
-                    className="apid-c-divider pf-u-pt-xl"
+                    className="apid-c-divider pf-u-pb-md"
                     inset={{default: 'insetNone',}}
                 />
+                <TextContent className="pf-u-pb-lg">
+                    <Text component={TextVariants.h2}>
+                      Principals
+                    </Text>
```

Comment:
I think, this part shouldn't be hard coded. This information needs to be inferred from the API docs.

Reply:
Yes, the principals is a tag or section from a particular API